datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -334,6 +334,100 @@ fn modulus_decimal(left: &DecimalArray, right: &DecimalArray) -> Result<DecimalA
     Ok(decimal_builder.finish())
 }
 
+/// The binary_bitwise_array_op macro only evaluates for integer types
+/// like int64, int32.
+/// It is used to do bitwise operation.
+macro_rules! binary_bitwise_array_op {
+    ($LEFT:expr, $RIGHT:expr, $OP:tt, $ARRAY_TYPE:ident, $TYPE:ty) => {{
+        let len = $LEFT.len();
+        let left = $LEFT.as_any().downcast_ref::<$ARRAY_TYPE>().unwrap();
+        let right = $RIGHT.as_any().downcast_ref::<$ARRAY_TYPE>().unwrap();
+        let result = (0..len)
+            .into_iter()
+            .map(|i| {
+                if left.is_null(i) || right.is_null(i) {
+                    None
+                } else {
+                    Some(left.value(i) & right.value(i))
```

Comment:
As below, I think this is meant to be `$OP`

```suggestion
                    Some(left.value(i) $OP right.value(i))
```

Reply:
Changed in https://github.com/apache/arrow-datafusion/pull/1653/commits/f81bffb326609be62e7468073ddc69f22171b69f (I was trying to make a follow on PR but then I found the logical conflict so I figured I would fix it directly on this PR). 

I hope that is OK with you

Reply:
This is from my mistake.